src/cplscheme/CouplingData.hpp

```diff
@@ -25,44 +46,52 @@ struct CouplingData { // @todo: should be a class from a design standpoint. See
     return data->values();
   }
 
-  /// Data values of previous iteration (1st col) and previous time windows.
-  DataMatrix oldValues;
+  void storeIteration()
+  {
+    _lastIteration = this->values();
+  }
 
-  mesh::PtrData data;
+  const Eigen::VectorXd lastIteration() const
+  {
+    return _lastIteration;
+  }
 
-  mesh::PtrMesh mesh;
+  [[deprecated("lastIteration should be read only!")]] void writeLastIteration(Eigen::VectorXd value)
+  {
+    _lastIteration = value;
+  }
 
-  ///  True, if the data values if this CouplingData requires to be initialized by a participant.
-  bool requiresInitialization;
+  int getMeshID()
+  {
+    return mesh->getID();
+  }
 
-  int getDimensions()
+  std::vector<int> getVertexOffsets()
   {
-    PRECICE_ASSERT(data != nullptr);
-    return data->getDimensions();
+    return mesh->getVertexOffsets();
   }
 
+  ///  True, if the data values if this CouplingData requires to be initialized by a participant.
+  const bool requiresInitialization;
+
+private:
   /**
    * @brief Default constructor, not to be used!
    *
    * Necessary when compiler creates template code for std::map::operator[].
    */
   CouplingData()
+      : requiresInitialization(false)
   {
     PRECICE_ASSERT(false);
   }
 
-  CouplingData(
-      mesh::PtrData data,
-      mesh::PtrMesh mesh,
-      bool          requiresInitialization)
-      : data(data),
-        mesh(mesh),
-        requiresInitialization(requiresInitialization)
-  {
-    PRECICE_ASSERT(data != nullptr);
-    PRECICE_ASSERT(mesh != nullptr);
-    PRECICE_ASSERT(mesh.use_count() > 0);
-  }
+  /// Data values of previous iteration.
+  Eigen::VectorXd _lastIteration;
```

Comment:
I like `_previousIteration` actually better.

Reply:
I agree. Renamed it in dd75e064837846d3da99deed1f8ed12ea6e841db.